src/dbus.c

```diff
@@ -656,6 +656,63 @@ static struct notification *dbus_message_to_notification(const gchar *sender, GV
         return n;
 }
 
+void signal_length_propertieschanged()
+{
+
+        static unsigned int last_displayed = 0;
+        static unsigned int last_history = 0;
+        static unsigned int last_waiting = 0;
+
+        if (!dbus_conn)
+                return;
+
+        GVariantBuilder *builder = g_variant_builder_new(G_VARIANT_TYPE_ARRAY);
+        GVariantBuilder *invalidated_builder = g_variant_builder_new(G_VARIANT_TYPE("as"));
+
+        unsigned int displayed = queues_length_displayed();
+        unsigned int history = queues_length_history();
+        unsigned int waiting = queues_length_waiting();
+        bool properties_changed = false;
+
+        if (last_displayed != displayed)
```

Comment:
Do we want to signal only the variables that changed? It might make it harder for programs to use. It feels a bit weird to have a variable amount of changes in a single signal. Maybe signalling every variable separately would be better?

Reply:
Allright, then it's fine. Could you maybe share the code you are using to test it?

Reply:
Oh I see, you linked a PR.